src/main/java/seedu/exercise/logic/commands/statistic/StatsFactory.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    /**
+     * Generates and returns statistic for different chart type.
+     */
+    public Statistic generateStatistic() {
+        HashMap<String, Double> data;
+        if (chart.equals("linechart")) {
+
+            return generateLineChartStatistic();
+
+        } else if (chart.equals("piechart")) {
+
+            return generatePieChartStatistic();
+
+        } else { //barchart
+
+            return generateBarChartStatistic();
+
+        }
+    }
+
+    /**
+     * Generate statistic for line chart.
+     */
+    private Statistic generateLineChartStatistic() {
+        ArrayList<Date> dates = Date.getListOfDates(startDate, endDate);
+        ArrayList<Double> values;
+
+        if (category.equals("exercise")) {
+            values = exerciseCountByDate(getFilteredExercise(), dates);
+        } else {
+            values = caloriesByDate(getFilteredExercise(), dates);
+        }
+
+        return new Statistic(category, chart, startDate, endDate, datesToString(dates), values);
+    }
+
+    /**
+     * Generate statistic for bar chart.
+     */
+    private Statistic generateBarChartStatistic() {
+        HashMap<String, Double> data;
+        if (category.equals("exercise")) {
+            data = getTotalExerciseQuantity();
+        } else { //calories
+            data = getTotalCaloriesData();
+        }
+
+        ArrayList<String> names = hashMapNameToList(data);
+        ArrayList<Double> values = hashMapDoubleToList(data, names);
+
+        return new Statistic(category, chart, startDate, endDate, names, values);
+    }
+
+    /**
+     * Generate statistic for pie chart.
+     */
+    private Statistic generatePieChartStatistic() {
+        HashMap<String, Double> data;
+        if (category.equals("exercise")) {
+            data = getTotalExerciseCount();
+        } else { //calories
+            data = getTotalCaloriesData();
+        }
+
+        ArrayList<String> names = hashMapNameToList(data);
+        ArrayList<Double> values = hashMapDoubleToList(data, names);
+
+        return new Statistic(category, chart, startDate, endDate, names, values);
+    }
+
+    /**
+     * Compute exercise count with filtered exercises list.
+     */
+    private HashMap<String, Double> getTotalExerciseCount() {
+        ArrayList<Exercise> filteredExercise = getFilteredExercise();
+        HashMap<String, Double> data = new HashMap<>();
+
+        for (Exercise e : filteredExercise) {
+            Name name = e.getName();
+            Unit unit = e.getUnit();
+            String nameWithUnit = name.toString() + " (" + unit.toString() + ")";
+            double count = 1;
+
+            if (data.containsKey(nameWithUnit)) {
+                count = data.get(nameWithUnit) + 1;
+            }
+
+            data.put(nameWithUnit, count);
+        }
+
+        return data;
+    }
+
+    /**
+     * Compute exercise quantity with filtered exercises list.
+     */
+    private HashMap<String, Double> getTotalExerciseQuantity() {
```

Comment:
Naming of this method (and getTotalExerciseCount) can be misleading.

Reply:
i have changed it to `getTotalExerciseFrequency()`